02-wrangling.Rmd

```diff
@@ -256,55 +320,182 @@ lang_no_delimiter <- separate(lang_messy_longer,
+to numeric data types in the `official_langs` data set as described in figure 
+\@ref(fig:img-mutate):
+
+```{r}
+official_langs <- mutate(official_langs,
   most_at_home = as.numeric(most_at_home),
   most_at_work = as.numeric(most_at_work)
 )
-lang_no_delimiter
+official_langs
 ```
 Now we see `<dbl>` appears under our columns, `most_at_home` and `most_at_work`, indicating they are double data types (which is one of the sub-types of numeric)!
 
-Is this data now tidy? Well, if we recall the three criteria for tidy data:
+Now when we filter we get what we expect: 
+```{r}
+filter(official_langs, most_at_home > 2669195)
+```
+
+In this case, we see that `filter` returns a data frame with only one row; this indicates that only English in Toronto is reported by more people as their primary language at home than French in Montréal.
+
+
+### Using `mutate` to create new columns
+
+```{r , include = FALSE}
+census_popn <- 35151728
+number_most_home <- filter(official_langs, language == "English" & region == "Toronto") |> 
+  pull(most_at_home)
+
+toronto_popn <- region_data |> filter(region == "Toronto") |> 
+  pull(population) 
+```
 
-- each row is a single observation,
-- each column is a single variable, and
-- each value is a single cell.
 
-We can see that this data now satisfies all three criteria, making it easier to analyze.
-For example, we could visualize how many people speak each of Canada's two 
+We can see in the table that `r format(number_most_home, scientific = FALSE, big.mark = ",")` people reported speaking English in Toronto as their primary language at home according to the 2016 Canadian census. What does this number mean to us? To understand this number, we need context. In particular, how many people were in Toronto when this data was collected? From the 2016 Canadian census profile, the population of Toronto was reported to be `r format(toronto_popn, scientific = FALSE, big.mark = ",")` people. The number of people who report that English as their primary language at home is much more meaningful when we report it in this context. We can even go a step further and transform this count to a relative frequency or proportion. We can do this by dividing the number of people reporting a given language as their primary language at home by the number of people who live in Toronto. For example, the proportion of people who reported that their primary language at home was English in the 2016 Canadian census was `r format(round(number_most_home/toronto_popn, 2), scientific = FALSE, big.mark = ",")` in Toronto.
+
+We can see the populations of the five cities by filtering the data
+using the `%in%` operator. The `%in%` operator is used to see if an element 
+belongs to a vector. Here we are filtering if any of the five cities 
+("Toronto", "Montréal", "Vancouver", "Calgary", "Edmonton") match the rows in the `region`
+column.
+```{r}
+filter(region_data, region %in% c("Toronto", "Montréal", "Vancouver", "Calgary", "Edmonton")) 
+```
```

Comment:
I feel like this should come in the `filter` section ... Especially when we want to reference backwards...

Reply:
Also, we need to teach `c()` somewhere... In the vector section above?

Reply:
Yup! I already added an explanation of `c()` at the beginning of the chapter when we talk about vectors -- is that fine or did you want me to expand on it more or move it? 

Good point about %in% -- will add to this section